src/components/TaskList/TaskList.jsx

```diff
@@ -87,63 +122,82 @@ class TaskList extends React.Component {
-    }
-    this.focusTaskAtIndex(newIndex)
-  }
-
-  focusPreviousTask() {
-    const oldIndex = this.state.selectedIndex
-    const lastIndex = this.props.tasks.length - 1
-    let newIndex = typeof oldIndex === 'number' ? oldIndex - 1 : lastIndex
-    if (newIndex < 0) {
-      newIndex = lastIndex
+  navRight() {
+    if (this.state.showHidden) {
+      return (
+        <div className="nav-right">
+          <div className="nav-right">
+            <span className="nav-item">
+              <button
+                type="button"
+                onClick={() => this.restore()}
+                className="control button is-link"
+                id="restore-button"
+                title="Restore selected"
+                disabled={!this.focusedTask()}
+              >↩️ Restore</button>
+            </span>
+          </div>
+        </div>
+      )
     }
-    this.focusTaskAtIndex(newIndex)
-  }
 
-  focusTaskAtIndex(index) {
-    console.info('focus task', this.props.tasks[index].storageKey)
-    this.setState({ selectedIndex: index })
+    return (
+      <div className="nav-right">
+        <span className="nav-item compact-vertically">
+          <button
+            type="button"
+            onClick={() => this.snooze()}
+            className="control button is-link"
+            id="snooze-button"
+            title="Snooze selected"
+            disabled={!this.focusedTask()}
+          >😴 Snooze</button>
+        </span>
+        <span className="nav-item compact-vertically">
+          <button
+            type="button"
+            id="archive-button"
+            className="control button is-link"
+            onClick={() => this.archive()}
+            title="Archive selected"
+            disabled={!this.focusedTask()}
+          >📥 Archive</button>
+        </span>
+        <span className="nav-item compact-vertically">
+          <button
+            type="button"
+            className="control button is-link"
+            onClick={() => this.ignore()}
+            title="Ignore selected"
+            disabled={!this.focusedTask()}
+          >❌ Ignore</button>
+        </span>
+      </div>
+    )
   }
 
   taskListOrMessage() {
-    const sortedTasks = this.props.tasks.sort((a, b) => {
-      return Date.parse(b.updatedAt) - Date.parse(a.updatedAt)
-    })
-
-    if (this.props.tasks.length > 0) {
+    if (this.tasks().length > 0) {
       return (
         <ol className="task-list">
-          {sortedTasks.map((task, index) => {
-            const isFocused = index === this.state.selectedIndex
-            const key = `${task.storageKey}-${task.isSelected}-${isFocused}`
-            return <TaskListItem task={task} key={key} isFocused={isFocused} />
+          {this.tasks().map((task, index) => {
+            const isFocused = (index === this.state.focusedIndex)
+            return (
+              <TaskListItem
+                task={task}
+                key={task.storageKey}
+                isFocused={isFocused}
```

Comment:
If you wanted, you could drop the curly braces on the `map` block, drop the `isFocused` variable in favor of doing the logic inline here, and drop the `return`:

```javascript
{this.tasks().map((task, index) => (
  <TaskListItem
```